prestoadmin/server.py

```diff
@@ -125,6 +127,15 @@ def uninstall():
         print('Package uninstalled successfully on: ' + env.host)
 
 
+@task
+@requires_topology
+def upgrade(local_package_path, local_config_dir):
+    configure_cmds.gather_directory(local_config_dir)
```

Comment:
instead of gathering the directory content to the presto-admin node and later redeploying it, it would be more efficient to copy the configuration to a temporary location on each node.